gr-blocks/lib/socket_pdu_impl.cc

```diff
@@ -29,6 +29,12 @@
 #include <gnuradio/io_signature.h>
 #include <gnuradio/blocks/pdu.h>
 
+#if BOOST_VERSION >= 107000
+#define GET_IO_SERVICE(s) ((boost::asio::io_context&)(s)->get_executor().context())
+#else
+#define GET_IO_SERVICE(s) ((s)->get_io_service())
+#endif
```

Comment:
Recommended in https://github.com/boostorg/asio/commit/59066d80b26e1d5b83b60d127ee17948d9ae9702#commitcomment-33184421
The 1.70 cast is potentially unsafe